src/ai/backend/common/pydantic_handlers.py

```diff
@@ -0,0 +1,254 @@
+
+    def get_all(self) -> dict[str, Any]:
+        return self.params
+
+
+async def _pydantic_handler(request: web.Request, handler) -> web.Response:
+    signature = inspect.signature(handler)
+    handler_params = _HandlerParameters()
+    for name, param in signature.parameters.items():
+        # Raise error when parameter has no type hint or not wrapped by 'Annotated'
+        if param.annotation is inspect.Parameter.empty:
+            raise InvalidAPIParameters(
+                f"Type hint or Annotated must be added in API handler signature: {param.name}"
+            )
+
+        value = await _extract_param_value(request=request, input_param_type=param.annotation)
+
+        if not value:
+            raise InvalidAPIParameters(
+                f"Type hint or Annotated must be added in API handler signature: {param.name}"
+            )
+
+        handler_params.add(name, value)
+
+    response = await handler(**handler_params.get_all())
+
+    if not isinstance(response, BaseResponse):
+        raise InvalidAPIParameters(
+            f"Only Response wrapped by BaseResponse Class can be handle: {type(response)}"
+        )
+
+    return web.json_response(response.data.model_dump(mode="json"), status=response.status_code)
+
+
+def pydantic_api_handler(handler):
+    @functools.wraps(handler)
+    async def wrapped(request: web.Request, *args, **kwargs) -> web.Response:
+        return await _pydantic_handler(request, handler)
+
+    return wrapped
+
+
+"""
+This decorator processes HTTP request parameters using Pydantic models.
+
+1. Request Body:
+    @pydantic_api_handler
+    async def handler(body: BodyParam[UserModel]):  # UserModel is a Pydantic model
+        user = body.parsed                          # 'parsed' property gets pydantic model you defined
+        return BaseResponse(data=YourResponseModel(user=user.id))
+
+2. Query Parameters:
+    @pydantic_api_handler
+    async def handler(query: QueryParam[QueryPathModel]):
+        parsed_query = query.parsed
+        return BaseResponse(data=YourResponseModel(search=parsed_query.query))
+
+3. Headers:
+    @pydantic_api_handler
+    async def handler(headers: HeaderParam[HeaderModel]):
+        parsed_header = headers.parsed
+        return BaseResponse(data=YourResponseModel(data=parsed_header.token))
+
+4. Path Parameters:
+    @pydantic_api_handler
+    async def handler(path: PathModel = PathParam(PathModel)):
+        parsed_path = path.parsed
+        return BaseResponse(data=YourResponseModel(path=parsed_path))
+
+5. Middleware Parameters:
+    # Need to extend MiddlewareParam and implement 'from_request'
+    class AuthMiddlewareParam(MiddlewareParam):
+        user_id: str
+        user_email: str
+        @classmethod
+        def from_request(cls, request: web.Request) -> Self:
+            # Extract and validate data from request
+            user_id = request["user"]["uuid"]
+            user_email = request["user"]["email"]
```

Comment:
This comment should be added as a comment in the function that will be used as a decorator.